signed/verifiers.go

```diff
@@ -137,3 +141,86 @@ func (v RSAPSSX509Verifier) Verify(key data.Key, sig []byte, msg []byte) error {
 
 	return verifyPSS(cert.PublicKey, digest[:], sig)
 }
+
+// ECDSAX509Verifier checks ECDSA signatures, extracting the public key
+// from an X509 certificate.
+type ECDSAX509Verifier struct{}
+
+// Verify does the actual check.
+func (v ECDSAX509Verifier) Verify(key data.Key, sig []byte, msg []byte) error {
+	digest := sha256.Sum256(msg)
+
+	k, _ := pem.Decode([]byte(key.Public()))
+	if k == nil {
+		logrus.Infof("Failed to decode PEM-encoded x509 certificate for keyID: %s", key.ID())
+		logrus.Debugf("Certificate bytes: %s", string(key.Public()))
+		return ErrInvalid
+	}
+	cert, err := x509.ParseCertificate(k.Bytes)
+	if err != nil {
+		logrus.Infof("Failed to parse x509 certificate: %s\n", err)
+		return ErrInvalid
+	}
+
+	ecdsaPubKey, ok := cert.PublicKey.(*ecdsa.PublicKey)
+	if !ok {
+		logrus.Infof("Value was not an RSA public key")
+		return ErrInvalid
+	}
+
+	sigLength := len(sig)
```

Comment:
This line to the end of the function appears to be replicated in the ECDSAVerifier. Would be good to refactor it out but not critical (if it works, it's never going to change)